Source/CarthageKit/DependencyCrawler.swift

```diff
@@ -0,0 +1,226 @@
+import Foundation
+import Result
+import ReactiveSwift
+
+/**
+Signals for DependencyCrawler events
+*/
+public enum DependencyCrawlerEvent {
+	case foundVersions(versions: [PinnedVersion], dependency: Dependency, versionSpecifier: VersionSpecifier)
+	case foundTransitiveDependencies(transitiveDependencies: [(Dependency, VersionSpecifier)], dependency: Dependency, version: PinnedVersion)
+	case failedRetrievingTransitiveDependencies(error: Error, dependency: Dependency, version: PinnedVersion)
+	case failedRetrievingVersions(error: Error, dependency: Dependency, versionSpeficier: VersionSpecifier)
```

Comment:
We shouldn't use `Error` directly—we should use a concrete error type.

Reply:
Now uses CarthageError